### PR TITLE
adding az logic, adding function to load launch_config

### PR DESCRIPTION
**Pull request by aaron-christian-nttd**
- Removing outdated README information
- Moving loading the launch_config to its own function. 
   - Updating references where this was being loaded from
   - Adding tests for functions
- Adding logic to run appropriate commands depending on provider. 
- Changing `aws_deployment_region` to `deployment_region` and moving this options to the _init_ as it is more common. 
- Implementing `uuid` back in launch_config to create if the `--skip_uuid`  is not set. 
- adding additional file list to the `launch service clean`. will now delete:
  - `.repo` folder
  - `components` folder
  - `.pre-commit-config.yaml` file
  
- Updating so the single source of version is `pyproject.toml`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,218 +97,32 @@ The service family of commands represents all the automation needed for the life
 
 `launch service create`: command is used to create a new service for the launch platform. This will create a new repository to the standard needed to run on the launch platform. This will be a properties driven only repository.
```
Comment (debasish-sahoo-nttd) [inline README.md:99]: any particular reason to delete the documentation?

File: :1
Comment (bryce-lynn-nttd) [inline :1:1]: Is this file here by accident?

**Review by bryce-lynn-nttd — APPROVED** [review]
I think that ":1" file is probably there by accident, but the rest looks ok to me.

**Review by kevin11-allen-nttd — APPROVED** [review]

File: src/launch/cli/service/create/__init__.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ def create(
         repository=repository,
         input_data=input_data,
         git_message=git_message,
-        uuid=not skip_uuid,
+        uuid=skip_uuid,
```
Comment (chris11-taylor-nttd) [inline src/launch/cli/service/create/__init__.py:216]: Seems like a strange change to make in isolation. Are we fixing the code to match the behavior laid out in the documentation?

File: src/launch/constants/version.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
+from importlib.metadata import version 
 from semver import Version
 
-VERSION = "0.13.0"
+
+VERSION = version('launch-cli')
```
Comment (chris11-taylor-nttd) [inline src/launch/constants/version.py:5]: Very happy to see this finally getting some traction!

File: src/launch/cli/terragrunt/__init__.py [hunk 4]
```diff
@@ -254,18 +262,27 @@ def terragrunt(
...
     install_tool_versions()
+    if IS_PIPELINE:
+        git_config(
+            dry_run=dry_run
+        )
 
-    # If the Provider is AZURE there is a prequisite requirement of logging into azure
-    # i.e. az login, or service principal is already applied to the environment.
+    # If the Provider is AZURE there is a prequisite requirement of logging into azure 
+    # i.e. az login, or service principal is already applied to the environment.: az login; az account get-access-token
```
Comment (chris11-taylor-nttd) [inline src/launch/cli/terragrunt/__init__.py:275]:
```suggestion
    # If the Provider is AZURE there is a prequisite requirement of logging into azure 
    # i.e. az login, or service principal is already applied to the environment: az login; az account get-access-token
```

File: src/launch/lib/automation/provider/az/functions.py [hunk 5]
```diff
@@ -48,3 +45,5 @@ def deploy_remote_state(
         subprocess.run(run_list, check=True)
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/automation/provider/az/functions.py:45]: For what it's worth, we can probably cut down on some of the path changing with the use of the `cwd` kwarg on subprocess.run. Don't need to change anything for this PR, but going forward I think that is a much cleaner solution that doesn't result in unexpected directory changes if there's an unhandled exception somewhere other than in the subprocess call.

File: src/launch/lib/service/build/functions.py [hunk 6]
```diff
@@ -7,19 +8,31 @@
...
     functions.make_build(dry_run=dry_run)
 
     if push:
+        click.secho(
+            "[TEST] in push.", fg="yellow"
+        )
         if registry_type == "docker":
-            functions.make_docker_aws_ecr_login(dry_run=dry_run)
+            click.secho(
+                "[TEST] Register docker.", fg="yellow"
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/service/build/functions.py:33]: ?

File: src/launch/lib/service/common.py [hunk 7]
```diff
@@ -70,3 +70,26 @@ def determine_existing_uuid(
...
+        Exception: If the launchconfig file cannot be read
+    """
+    with open(path, "r") as f:
+        click.secho(
+            f"Reading config file found at local path: {LAUNCHCONFIG_PATH_LOCAL=}"
+        )
+        input_data = json.load(f)
+        input_data = input_data_validation(input_data)
+        return input_data
+    raise Exception("Failed to read launchconfig file")
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/service/common.py:95]: The final `raise` statement here doesn't seem like it's in a reachable code path. The `open()` context manager will raise a FileNotFound exception, for example, when provided a filename that doesn't exist. What was the intent here?

File: src/launch/lib/service/template/launchconfig.py [hunk 8]
```diff
@@ -34,26 +34,28 @@ def properties_file(self, value: dict, current_path: Path, dest_base: Path) -> N
...
-                target_path.parent.mkdir(parents=True, exist_ok=True)
-                shutil.copy(src=source_path, dst=target_path)
-                click.echo(
-                    f"Copied additional file from {source_path} to {target_path}"
-                )
+                try:
+                    target_path.parent.mkdir(parents=True, exist_ok=True)
+                    shutil.copy(file_path, target_path)
+                except shutil.SameFileError:
+                    pass
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/service/template/launchconfig.py:58]: This is acceptable as-is, but I wanted to provide you with an alternative for the future that I find to be a lot more readable:

```python
from contextlib import suppress

with suppress(shutil.SameFileError):
    # your code that might raise a SameFileError that you wish to ignore
```

**Review by debasish-sahoo-nttd — APPROVED** [review]

**Review by chris11-taylor-nttd — APPROVED** [review]

**Pull request merged**